### PR TITLE
Bootstrap feature. rke dind ATT. namespace move. Docs

### PR DESCRIPTION
**Pull request by rawmind0**
This PR includes:
- Updated dapper scripts to include acceptance tests with rke dind. 
- Added RANCHER_ACC_CLUSTER_NAME env variable to get dynamic cluster name for ATT.
- Removed config file on provider. 
- Added util functions.
- Added resource rancher2 setting. 
- Added namespace move between projects.
- Added insecure option on config.
- Added pagination on rancher2 list calls.
- Updated README and website docs.
- Updated resource node-driver messages and tests.

### CODE REVIEW

File: scripts/test [hunk 1]
```diff
@@ -1,10 +1,8 @@
 #!/bin/bash
 set -e
 
-cd $(dirname $0)/..
-
 echo Running tests
 
-PACKAGES="$(find . -name '*.go' | xargs -I{} dirname {} |  cut -f2 -d/ | sort -u | grep -Ev '(^\.$|.git|.trash-cache|vendor|bin)' | sed -e 's!^!./!' -e 's!$!/...!')"
+PACKAGES="$(find .. -name '*.go' | xargs -I{} dirname {} |  cut -f2 -d/ | sort -u | grep -Ev '(^\..$|.git|.trash-cache|vendor|bin)' | sed -e 's!^!../!' -e 's!$!/...!')"
```
Comment (cloudnautique) [inline scripts/test]: this seems like an odd change, whats the motivation?

File: scripts/testacc [hunk 2]
```diff
@@ -7,17 +7,17 @@ cleanup()
...
 RANCHER_SECRET_KEY=${RANCHER_SECRET_KEY:-""}
-
-cd $(dirname $0)/..
+RANCHER_INSECURE=${RANCHER_INSECURE:-"false"}
+RANCHER_ACC_CLUSTER_NAME=${RANCHER_ACC_CLUSTER_NAME:-"local"}
 
 echo Running acceptance tests
 
-PACKAGES="$(find . -name '*.go' | xargs -I{} dirname {} |  cut -f2 -d/ | sort -u | grep -Ev '(^\.$|.git|.trash-cache|vendor|bin)' | sed -e 's!^!./!' -e 's!$!/...!')"
+PACKAGES="$(find .. -name '*.go' | xargs -I{} dirname {} |  cut -f2 -d/ | sort -u | grep -Ev '(^\..$|.git|.trash-cache|vendor|bin)' | sed -e 's!^!../!' -e 's!$!/...!')"
```
Comment (cloudnautique) [inline scripts/testacc]: why .. ??

Comment (rawmind0) [reply scripts/testacc]: As script was doing `cd $(dirname $0)/..` cleanup() was failing to execute on some circunstances. 

I removed `cd $(dirname $0)/..`, adapted packages find to be executed from current path and cleanup() is working fine.

File: scripts/test (hunk 1 above)
Comment (rawmind0) [reply scripts/test]: As i needed to change it at `scripts/testacc`, i also changed it here to let them in the same format. Take a look to the next comment.

Comment (cloudnautique) [reply scripts/test]: I would go back to changing directory back to the root.

File: rancher2/config.go [hunk 3]
```diff
@@ -132,12 +251,19 @@ func (c *Config) GetProjectRoleTemplateBindingsByProjectID(projectID string) ([]
...
+	collection, err := client.ProjectRoleTemplateBinding.List(listOpts)
 	if err != nil {
 		return nil, err
 	}
 
-	return projectsRoles.Data, nil
+	// Paginating data if needed
+	data := collection.Data
+
+	for collection, err = collection.Next(); err != nil && collection != nil; collection, err = collection.Next() {
```
Comment (cloudnautique) [inline rancher2/config.go]: This is kind of confusing and hard to follow with variable names as written. Is there a way to check if there is a next? Or can we query with a limit opt like 100 and `if len(collection.Data ==100) { check next }` that way you don't have to make an unnecessary HTTP call.

File: rancher2/config.go [hunk 4]
```diff
@@ -153,12 +279,19 @@ func (c *Config) GetProjectByName(name, clusterID string) (*managementClient.Pro
...
-	for _, project := range projects.Data {
+	// Paginating data if needed
+	data := collection.Data
+
+	for collection, err = collection.Next(); err != nil && collection != nil; collection, err = collection.Next() {
+		data = append(data, collection.Data...)
+	}
+
+	for _, project := range data {
 		if project.Name == name {
```
Comment (cloudnautique) [inline rancher2/config.go]: Are project names unique across a cluster? If not, you could end up ignoring a 2nd, 3rd, 4th. If they are unique, you should be able to get away with a simple:
`if len(collection.Data) > 0 { return collection.Data[0] }`

Really, we should be operating by Ids where/when ever possible.

File: rancher2/config.go [hunk 5]
```diff
@@ -248,12 +381,19 @@ func (c *Config) GetClusterByName(name string) (*managementClient.Cluster, error
...
-	for _, cluster := range clusters.Data {
+	// Paginating data if needed
+	data := collection.Data
+
+	for collection, err = collection.Next(); err != nil && collection != nil; collection, err = collection.Next() {
+		data = append(data, collection.Data...)
+	}
+
+	for _, cluster := range data {
 		if cluster.Name == name {
```
Comment (cloudnautique) [inline rancher2/config.go]: Same as above, are cluster names globally unique? I don't know that they are.

If they aren't, then you could have more then 1 and miss the users intended. Also, you don't return the data slice, so there really isn't a need to build it... you could do something around limits and determine when to call next intelligently without making extra http round trips.

File: scripts/testacc (hunk 2 above)
Comment (cloudnautique) [reply scripts/testacc]: hmm. I think it would be better to work out the dirname method.

File: scripts/test (hunk 1 above)
Comment (rawmind0) [reply scripts/test]: Done, going back to changing directory back to the root

File: scripts/testacc (hunk 2 above)
Comment (rawmind0) [reply scripts/testacc]: Changed dirname method for `cleaup()` function and going back to changing directory back to the root

File: rancher2/config.go (hunk 3 above)
Comment (rawmind0) [reply rancher2/config.go]: Agreed. Checking if `collection.Pagination.Partial == true` to paginating

File: rancher2/config.go (hunk 4 above)
Comment (rawmind0) [reply rancher2/config.go]: Project names are not unique across cluster. Changed to return first and ignoring a 2nd, 3rd, 4th.

File: rancher2/config.go (hunk 5 above)
Comment (rawmind0) [reply rancher2/config.go]: Right, cluster names are globally unique. Changing to `if len(collection.Data) > 0 { return collection.Data[0], nil }`

**Comment by cloudnautique** [comment]
I bootstrapped a Rancher install. Couple of items:

1. Can we provide more feedback when the admin password is set.
2. Can we clean up the token as part of the run? Is there a post hook or something we can tie into in TF?
3. how would you change the admin password?

**Comment by rawmind0** [comment]
@cloudnautique answering items:

1. Yes, could try to provide more feedback about it.
2. Just last token with default TTL is not removed. I'll try to figure out if a post hook could be called.
3. Admin password could be changed just if bootstrap isn't done. I could add a resource `Rancher2_admin_config` to do that.

**Comment by cloudnautique** [comment]
@rawmind0, for 3 I don't think that will work well, because the admin password is set in 2 places? Also, if a Random password is used, is the admin password an output? I suppose if the password were stored in the state file you could use that one, and update to the new? It would also allow for exporting.

File: rancher2/config.go (hunk 4 above)
Comment (cloudnautique) [reply rancher2/config.go]: OK, I think this is going to cause problems. If we fetch by name, we will only ever be able to use 2nd, 3rd or N project which is not desirable. We need to manage IDs only.

**Comment by rawmind0** [comment]
@cloudnautique additional comments on items:
1. `admin_pass` is just set if bootstrap wasn't done before, and login with default user/pass is possible. Do you mean to show a log or warn message??
2. AFAIK, there is no chance to set a hook to remove all tokens. Right now, just the last generated token remains created with TTL, and before generates a new one admin token cleanup is done, to remove older tokens.
3. i mean change to use `rancher2_admin_config` instead `admin_pass` on tf provider config. Using `admin_pass` random password option is not available, but using `rancher2_admin_config` could be done because admin password would be stored on tfstate file. But the problem with that option is that you need to edit provider config once you've changed admin password. In that point, i was trying with both options, `rancher2_admin_config` and `admin_pass`, and may be `admin_pass` is a little bit cleaner.

**Comment by rawmind0** [comment]
@cloudnautique , for point 3 i added `admin_new_pass` provider argument just to provide the option to change admin password. As it's not a resource and not saved in tfstate file, argument should be remove once password is changed.

**Comment by cloudnautique** [comment]
will it break if the new password argument is not removed? It should do the right thing?

**Comment by rawmind0** [comment]
It will not break anything, but
- if using token to auth, it will try to set the new password again. 
- If using `admin_pass` argument to auth and not updated, auth will fail on next execution.

**Comment by rawmind0** [comment]
@cloudnautique , updated PR removing bootstrap action, `admin_pass` and `admin_new_pass` arguments.

**Comment by rawmind0** [comment]
Includes insecure option like PR https://github.com/rancher/terraform-provider-rancher2/pull/15 and namespace moving like PR https://github.com/rancher/terraform-provider-rancher2/pull/10
